extensions/interactions/RatioExpressionInput/directives/oppia-interactive-ratio-expression-input.component.ts

```diff
@@ -17,112 +17,133 @@
-    function(
-        $attrs, $scope, CurrentInteractionService,
-        InteractionAttributesExtractorService,
-        RatioExpressionInputRulesService) {
-      var ctrl = this;
-      var errorMessage = '';
-      // Label for errors caused whilst parsing ratio expression.
-      var FORM_ERROR_TYPE = 'RATIO_EXPRESSION_INPUT_FORMAT_ERROR';
-      ctrl.getWarningText = function() {
-        return errorMessage;
-      };
+  ngOnInit(): void {
+    const {
+      placeholder,
+      numberOfTerms
+    } = this.interactionAttributesExtractorService.getValuesFromAttributes(
+      'RatioExpressionInput',
+      {
+        placeholderWithValue: this.placeholderWithValue,
+        numberOfTermsWithValue: this.numberOfTermsWithValue
+      }
+    ) as RatioExpressionInputCustomizationArgs;
+    this.expectedNumberOfTerms = numberOfTerms.value;
+    this.placeholder = placeholder.value.unicode;
+    if (this.savedSolution !== undefined) {
+      let savedSolution = this.savedSolution;
+      savedSolution = Ratio.fromList(
+        savedSolution as RatioInputAnswer).toAnswerString();
+      this.answer = savedSolution;
+    } else {
+      this.answer = '';
+    }
+    const submitAnswerFn = () => this.submitAnswer();
+    const isAnswerValid = () => this.isAnswerValid();
+    this.currentInteractionService.registerCurrentInteraction(
+      submitAnswerFn, isAnswerValid);
+
+    setTimeout(
+      () => {
+        let focusLabel: string = this.labelForFocusTarget;
+        this.focusManagerService.setFocusWithoutScroll(focusLabel);
+      }, 0);
+  }
 
-      ctrl.submitAnswer = function(answer) {
-        try {
-          var ratioExpression =
-            Ratio.fromRawInputString(answer);
-          if (
-            (
-              ratioExpression.getNumberOfTerms() !==
-              ctrl.expectedNumberOfTerms) && (ctrl.expectedNumberOfTerms !== 0)
-          ) {
-            throw new Error(
-              'The creator has specified the number of terms in' +
-              ' the answer to be ' + ctrl.expectedNumberOfTerms + '.');
-          }
-          errorMessage = '';
-          ctrl.RatioExpressionInputForm.answer.$setValidity(
-            FORM_ERROR_TYPE, true);
-          CurrentInteractionService.onSubmit(
-            ratioExpression.getComponents(),
-            RatioExpressionInputRulesService);
-        } catch (parsingError) {
-          errorMessage = parsingError.message;
-          ctrl.RatioExpressionInputForm.answer.$setValidity(
-            FORM_ERROR_TYPE, false);
-        }
-      };
+  submitAnswer(): void {
+    try {
+      const ratioExpression =
+        Ratio.fromRawInputString(this.answer);
+      if (
+        (
+          ratioExpression.getNumberOfTerms() !==
+          this.expectedNumberOfTerms) && (this.expectedNumberOfTerms !== 0)
+      ) {
+        throw new Error(
+          'The creator has specified the number of terms in' +
+          ' the answer to be ' + this.expectedNumberOfTerms + '.');
```

Comment:
```suggestion
          `The creator has specified the number of terms in `
          `the answer to be ${this.expectedNumberOfTerms}.`
        );
```

Reply:
ig it should be in single quotes, as line 118 still in string.

Reply:
Not sure what you mean.

Reply:
![Screenshot from 2022-01-19 23-42-27](https://user-images.githubusercontent.com/80846488/150189761-0fead05b-f8bd-4307-a1a1-6d0ec6bb08f8.png)

![Screenshot from 2022-01-19 23-46-05](https://user-images.githubusercontent.com/80846488/150190223-7b48e0b7-05b8-4796-995c-1632b5cfb0e4.png)

now you can understand what i am trying to say.

Reply:
Hmm, interesting. Can you use single on the first line and backtick on the second line?

Reply:
Done